File: androidApp/src/main/java/io/redlink/more/app/android/shared_composables/AppVersion.kt
/*
 * Copyright LBI-DHP and/or licensed to LBI-DHP under one or more
 * contributor license agreements (LBI-DHP: Ludwig Boltzmann Institute
 * for Digital Health and Prevention -- A research institute of the
 * Ludwig Boltzmann Gesellschaft, Österreichische Vereinigung zur
 * Förderung der wissenschaftlichen Forschung).
 * Licensed under the Apache 2.0 license with Commons Clause
 * (see https://www.apache.org/licenses/LICENSE-2.0 and
 * https://commonsclause.com/).
 */
package io.redlink.more.app.android.shared_composables

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import io.redlink.more.app.android.BuildConfig
import io.redlink.more.app.android.R
import io.redlink.more.app.android.extensions.getStringResource
import io.redlink.more.app.android.ui.theme.MoreColors

@Composable
fun AppVersion() {
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.Center,
        modifier = Modifier.fillMaxWidth()
            .padding(vertical = 10.dp)
    ) {
        Text(
            text = "${getStringResource(R.string.app_version)}: ${BuildConfig.VERSION_NAME}",
            fontSize = 10.sp,
            color = MoreColors.Primary,
            fontWeight = FontWeight.Medium
        )
    }
}

@Composable
@Preview
fun AppVersionPreview() {
    AppVersion()
}

File: iosApp/iosApp/Views/Components/AppVersion.swift
//
//  AppVersion.swift
//  More
//
//  Created by Jan Cortiel on 08.08.23.
//  Copyright © 2023 Ludwig Boltzmann Institute for
//  Digital Health and Prevention - A research institute
//  of the Ludwig Boltzmann Gesellschaft,
//  Oesterreichische Vereinigung zur Foerderung
//  der wissenschaftlichen Forschung 
//  Licensed under the Apache 2.0 license with Commons Clause 
//  (see https://www.apache.org/licenses/LICENSE-2.0 and
//  https://commonsclause.com/).
//

import SwiftUI

struct AppVersion: View {
    var body: some View {
        Text("\("App Version".localize(withComment: "App Version")): \(Bundle.main.appBuild)")
            .font(.system(size: 10, weight: .medium))
            .foregroundColor(.more.primary)
            .padding(.vertical, 10)
    }
}

struct AppVersion_Previews: PreviewProvider {
    static var previews: some View {
        AppVersion()
    }
}
